plugins/org.eclipse.emf.mwe2.language/src/org/eclipse/emf/mwe2/language/validation/Mwe2Validator.java

```diff
@@ -399,4 +402,30 @@ public boolean isMandatory(JvmIdentifiableElement feature) {
 		return ePackages;
 	}
 
+	@Check
+	public void checkDeprecatedComponent(Component component) {
+		JvmType type = component.getType();
+		if (type instanceof JvmAnnotationTarget && DeprecationUtil.isDeprecated((JvmAnnotationTarget) type)) {
+			warning(
+				"The '" + type.getQualifiedName() + "' is deprecated.",
+				component,
+				Mwe2Package.Literals.REFERRABLE__NAME,
+				DEPRECATED_ELEMENT);
+		}
+	}
+
+	@Check
+	public void checkDeprecatedAssignment(Assignment assignment) {
+		JvmIdentifiableElement feature = assignment.getFeature();
+		if (feature instanceof JvmOperation) {
+			JvmOperation operation = (JvmOperation) feature;
+			if (DeprecationUtil.isDeprecated(operation)) {
```

Comment:
can and should this be used in the semantic highlighting too?

Reply:
Yes, I will create a follow up PR about that refactoring